packages/amazon/src/serverGroup/configure/wizard/instanceType/InstanceTypeSelector.tsx

```diff
@@ -0,0 +1,142 @@
+  setFieldValue: (field: keyof IAmazonServerGroupCommand, value: any, shouldValidate?: boolean) => void;
+  instanceTypeDetails: IInstanceTypeCategory[];
+}
+
+export function InstanceTypeSelector(props: IInstanceTypeSelectorProps) {
+  const isLaunchTemplatesEnabled = AWSProviderSettings.serverGroups?.enableLaunchTemplates;
+
+  const [useSimpleMode, setUseSimpleMode] = React.useState(props.command.viewState.useSimpleInstanceTypeSelector);
+  const [unlimitedCpuCredits, setUnlimitedCpuCredits] = useState(props.command.unlimitedCpuCredits);
+
+  useEffect(() => {
+    if (props.command.unlimitedCpuCredits !== unlimitedCpuCredits) {
+      props.setFieldValue('unlimitedCpuCredits', unlimitedCpuCredits);
+    }
+  }, [unlimitedCpuCredits]);
+
+  const handleModeChange = (useSimpleModeNew: boolean) => {
+    if (useSimpleMode !== useSimpleModeNew) {
+      setUseSimpleMode(useSimpleModeNew);
+
+      // update viewState
+      props.setFieldValue('viewState', {
+        ...props.command.viewState,
+        useSimpleInstanceTypeSelector: useSimpleModeNew,
+      });
+
+      // update selected instance type(s) if mode changed.
+      // Simple mode uses command.instanceType to track selected type. Advanced mode uses command.launchTemplateOverridesForInstanceType to track selected types.
+      const multipleInstanceTypesInProps = props.command.launchTemplateOverridesForInstanceType;
+      const singleInstanceTypeInProps = props.command.instanceType;
+
+      if (useSimpleModeNew && multipleInstanceTypesInProps && multipleInstanceTypesInProps.length > 0) {
+        // advanced mode -> simple mode, pick the instance type with highest priority from command.launchTemplateOverridesForInstanceType
+        const highestPriorityNum = Math.min(...multipleInstanceTypesInProps.map((it) => it.priority));
+        const instanceTypeWithHighestPriority = multipleInstanceTypesInProps.find(
+          (it) => it.priority === highestPriorityNum,
+        ).instanceType;
+
+        props.setFieldValue('instanceType', instanceTypeWithHighestPriority);
+        props.setFieldValue('launchTemplateOverridesForInstanceType', []);
+        props.command.instanceTypeChanged(props.command);
+      } else if (!useSimpleModeNew && singleInstanceTypeInProps) {
+        // simple mode -> advanced mode, port command.instanceType to command.launchTemplateOverridesForInstanceType
+        const instanceTypes: IAmazonInstanceTypeOverride[] = [
+          {
+            instanceType: singleInstanceTypeInProps,
+            priority: 1,
+          },
+        ];
+        props.setFieldValue('launchTemplateOverridesForInstanceType', instanceTypes);
+        props.setFieldValue('instanceType', undefined);
+        props.command.instanceTypesChanged(props.command);
+      }
+    }
+  };
+
+  // advanced mode is supported only when launch templates support is enabled, so, render the component iff launch templates support is enabled
+  if (isLaunchTemplatesEnabled && !useSimpleMode) {
```

Comment:
IMO this comment is a bit redundant. It would be more readable to have `const showAdvancedMode = isLaunchTemplatesEnabled && !useSimpleMode`

then: `if (showAdvancedMode)`